[ao] Added table visualization capability to ModelReportVisualizer

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81975
* #81974
* __->__ #81973
* #81673

Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Added table visualization capability to ModelReportVisualizer

Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/_model_report/model_report_visualizer.py:
```diff
@@ -376,7 +384,34 @@ def generate_table_visualization(self, feature_filter: str = "", module_fqn_filt
+        # now we have populated the tables for each one
+        # let's create the strings to be returned
+        table_str = ""
+        if len(tensor_headers) > self.DEFAULT_NON_FEATURE_TENSOR_HEADERS:
```
also I think it should be `NUM_NON_FEATURE_TENSOR_HEADERS` to be more accurate?

vspenubarthi replied:
changed and added a better description for what this check is

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Added table visualization capability to ModelReportVisualizer"


Summary: This adds the capability to visualize the table of information
in the ModelReportVisualizer. This allows the user to filter based on
module name pattern match or feature name pattern match and the
implemented method `generate_table_visualization` prints out the table
in a string format that is easy to parse.

Expected Usage
```
mod_rep_visualizer.generate_table_visualization()
```
Can also pass in optional filters as well if needed.

The tests for this were just visual inspection for two reasons:
1.) This method does not return anything, it just generates the
visualization
2.) All the data to create the table visualization is gotten from
`generate_filtered_tables` which is already tested, so testing all that
for this again would be redundant.

Example Printed Output
```
Tensor Level Information
  idx  layer_fqn        input_activation_global_max    input_activation_global_min    input_weight_channel_axis    input_weight_threshold    outlier_detection_channel_axis    outlier_detection_ratio_threshold    outlier_detection_reference_percentile    weight_global_max    weight_global_min
-----  -------------  -----------------------------  -----------------------------  ---------------------------  ------------------------  --------------------------------  -----------------------------------  ----------------------------------------  -------------------  -------------------
    1  block1.linear                        1.9543                        -1.33414                            1                       0.5                                 1                                  3.5                                      0.95             0.380521           -0.568476
    2  block2.linear                        1.81486                        0                                  1                       0.5                                 1                                  3.5                                      0.95             0.521438           -0.0256195

 Channel Level Information
  idx  layer_fqn        channel    constant_batch_counts    input_activation_per_channel_max    input_activation_per_channel_min    input_weight_channel_comparison_metrics  input_weight_equalization_recommended      outlier_detection_batches_used  outlier_detection_is_sufficient_batches      outlier_detection_percentile_ratios  outliers_detected      weight_per_channel_max    weight_per_channel_min
-----  -------------  ---------  -----------------------  ----------------------------------  ----------------------------------  -----------------------------------------  ---------------------------------------  --------------------------------  -----------------------------------------  -------------------------------------  -------------------  ------------------------  ------------------------
    1  block1.linear          0                        0                            1.9543                             -1.33414                                    0.956912  True                                                                    1  True                                                                     1.77489  False                                0.300502                -0.568476
    2  block1.linear          1                        0                            1.14313                            -0.756184                                   1.04378   True                                                                    1  True                                                                     2.07887  False                                0.336131                -0.261025
    3  block1.linear          2                        0                            0.653274                           -0.937748                                   1.10837   True                                                                    1  True                                                                     1.00712  False                                0.380521                -0.183536
    4  block2.linear          0                        0                            1.81486                             0                                          0.542731  True                                                                    1  True                                                                     1.78714  False                                0.13552                 -0.0256195
    5  block2.linear          1                        0                            1.72578                             0                                          0.505475  True                                                                    1  True                                                                     1.40475  False                                0.485536                 0.352621
    6  block2.linear          2                        0                            1.7284                              0                                          0.909304  True                                                                    1  True                                                                     1.40392  False                                0.521438                 0.0906605
```

Test Plan: Visual Test

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g
